src/Parser/DisciplinePriest/Modules/Items/InnerHallation.js

```diff
@@ -0,0 +1,46 @@
+import SPELLS from 'common/SPELLS';
+import ITEMS  from 'common/ITEMS';
+
+import Module from 'Parser/Core/Module';
+
+const debug = false;
+
+class InnerHallation extends Module {
+  manaGained = 0;
+
+  // timestamp for power infusion if its talented & casted (to exclude mana saved):
+  lastPowerInfusionCastStartTimestamp = null;
+
+  on_initialized() {
+    if (!this.owner.error) {
+      const selectedCombatant = this.owner.selectedCombatant;
+      this.active = selectedCombatant.hasShoulder(ITEMS.INNER_HALLATION.id);
+    }
+  }
+
+  on_byPlayer_cast(event) {
+    if(this.owner.selectedCombatant.hasTalent(SPELLS.POWER_INFUSION.id) && event.ability.guid === SPELLS.POWER_INFUSION.id){
+      this.lastPowerInfusionCastStartTimestamp = event.timestamp;
+      return;
+      
+    } else if (this.owner.selectedCombatant.hasBuff(SPELLS.POWER_INFUSION.id) && (event.timestamp + 20000) > this.lastPowerInfusionCastStartTimestamp){
```

Comment:
I thought it would be an performance increase but apparently it isn't. I will see what I can do about the time modifying mechanics.